### PR TITLE
[spirv] Add tests for half and short constants.

### PR DESCRIPTION
**Pull request by ehsannas**

### CODE REVIEW

File: tools/clang/test/CodeGenSPIRV/constant.scalar.hlsl [hunk 1]
```diff
@@ -61,4 +61,16 @@ void main() {
...
 // CHECK-DAG: %double_n1234567898765_32 = OpConstant %double -1234567898765.32
   double c_nlarge = -1234567898765.32;
+
+// Since we translate HLSL 'half' to 32-bit float in SPIR-V, float constants are generated.
+// CHECK-DAG: %float_7_7 = OpConstant %float 7.7
+  half c_half_4_5 = 7.7;
+// CHECK-DAG: %float_n8_8 = OpConstant %float -8.8
+  half c_half_n8_2 = -8.8;
+
+// Since we translate HLSL 'short' to 32-bit int in SPIR-V, integer constants are generated.
```
Comment (antiagainst) [inline tools/clang/test/CodeGenSPIRV/constant.scalar.hlsl]: We are not testing `short` type here?

File: tools/clang/test/CodeGenSPIRV/constant.scalar.hlsl [hunk 2]
```diff
@@ -61,4 +61,16 @@ void main() {
   double c_large  =  1234567898765.32;
 // CHECK-DAG: %double_n1234567898765_32 = OpConstant %double -1234567898765.32
   double c_nlarge = -1234567898765.32;
+
+// Since we translate HLSL 'half' to 32-bit float in SPIR-V, float constants are generated.
+// CHECK-DAG: %float_7_7 = OpConstant %float 7.7
+  half c_half_4_5 = 7.7;
```
Comment (antiagainst) [inline tools/clang/test/CodeGenSPIRV/constant.scalar.hlsl]: Also check `RelaxedPrecision` decoration?

File: tools/clang/test/CodeGenSPIRV/constant.scalar.hlsl [hunk 3]
```diff
@@ -61,4 +61,16 @@ void main() {
   double c_large  =  1234567898765.32;
 // CHECK-DAG: %double_n1234567898765_32 = OpConstant %double -1234567898765.32
   double c_nlarge = -1234567898765.32;
+
+// Since we translate HLSL 'half' to 32-bit float in SPIR-V, float constants are generated.
+// CHECK-DAG: %float_7_7 = OpConstant %float 7.7
```
Comment (antiagainst) [inline tools/clang/test/CodeGenSPIRV/constant.scalar.hlsl]: Hmm, I would suggest to convert all `CHECK-DAG` directives to `CHECK` directives in this file. `CHECK-DAG` takes a much longer time to run. The flexibility it gives us is not quite beneficial here.

**Comment by ehsannas** [comment]
Abandoning this CL. A more comprehensive change is required. New PR will be submitted.

**Pull request closed without merging**